tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,316 @@
+			string syncToken1 = "syncToken1;sn=001";
+			Uri uri1 = new Uri("https://store1.resource.io/kv/searchQuery1");
+			string eventType1 = "eventType.KeyValueModified";
+
+			string sampleMessage2 = "{\"id\":\"id-value2\",\"topic\":\"/subscriptions/subscription-value2/resourceGroups/resourceGroup2/providers/provider2/configurationstores/store2\",\"subject\":\"https://store2.resource.io/kv/searchQuery2\",\"data\":{\"key\":\"searchQuery2\",\"etag\":\"etagValue2\",\"syncToken\":\"syncToken2;sn=002\"},\"eventType\":\"eventType.KeyValueDeleted\",\"dataVersion\":\"2\",\"metadataVersion\":\"1\",\"eventTime\":\"2021-10-06T20:08:07.2536025Z\"}";
+			string syncToken2 = "syncToken2;sn=002";
+			Uri uri2 = new Uri("https://store2.resource.io/kv/searchQuery2");
+			string eventType2 = "eventType.KeyValueDeleted";
+
+			//Parse the sampleMessages into the pushNotification
+			EventGridEventParser.TryParseJson(sampleMessage1, out PushNotification pushNotification1);
+			Assert.Equal(pushNotification1.SyncToken, syncToken1);
+			Assert.Equal(pushNotification1.ResourceUri, uri1);
+			Assert.Equal(pushNotification1.EventType, eventType1);
+
+			EventGridEventParser.TryParseJson(sampleMessage2, out PushNotification pushNotification2);
+			Assert.Equal(pushNotification2.SyncToken, syncToken2);
+			Assert.Equal(pushNotification2.ResourceUri, uri2);
+			Assert.Equal(pushNotification2.EventType, eventType2);
+		}
+
+		[Fact]
+		public void PushNotification_TestNullPushNotificationProcessPushNotification()
+        {
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			//Create list of PushNotifications with null Parameters
+			List<PushNotification> invalidPushNotifications = invalidPushNotificationList;
+
+			foreach (PushNotification invalidPushNotification in invalidPushNotifications)
+            {
+                try
+                {
+                    refresher.ProcessPushNotification(invalidPushNotification);
+                    refresher.RefreshAsync().Wait();
+
+                }
+				//Should be caught in this block and continue to the next invalidNotification
+                catch (ArgumentNullException) { Assert.True(true); continue; }
+                catch (ArgumentException) { Assert.True(true); continue; }
+
+				//ProcessPushNotification did not throw any errors
```

Comment:
We shouldn't try/catch in unit tests - use `Assert.Throws` to verify which exception is thrown.